test/parallel/test-util-callbackify.js

```diff
@@ -0,0 +1,192 @@
+    async function asyncFn() {
+      return await Promise.reject(value);
+    }
+
+    const cbAsyncFn = callbackify(asyncFn);
+
+    cbAsyncFn(common.mustCall((err, ret) => {
+      assert.strictEqual(ret, undefined);
+      if (err instanceof Error) {
+        if ('reason' in err) {
+          assert.strictEqual(err.code, 'NULL_REJECTION');
+          assert.strictEqual(err.reason, value);
+        } else {
+          assert.strictEqual(String(value).endsWith(err.message), true);
+        }
+      } else {
+        assert.strictEqual(err, value);
+      }
+    }));
+  }
+
+  for (const value of values) {
+    function promiseFn() {
+      return Promise.reject(value);
+    }
+
+    const cbPromiseFn = callbackify(promiseFn);
+
+    cbPromiseFn(common.mustCall((err, ret) => {
+      assert.strictEqual(ret, undefined);
+      if (err instanceof Error) {
+        if ('reason' in err) {
+          assert(!value);
+          assert.strictEqual(err.code, 'NULL_REJECTION');
+          assert.strictEqual(err.reason, value);
+        } else {
+          assert.strictEqual(String(value).endsWith(err.message), true);
+        }
+      } else {
+        assert.strictEqual(err, value);
+      }
+    }));
+  }
+}
+
+{
+  // Test that arguments passed to callbackified function are passed to original
+  for (const value of values) {
+    async function asyncFn(arg) {
+      assert.strictEqual(arg, value);
+      return await Promise.resolve(arg);
+    }
+
+    const cbAsyncFn = callbackify(asyncFn);
+
+    cbAsyncFn(value, common.mustCall((err, ret) => {
+      assert.ifError(err);
+      assert.strictEqual(ret, value);
+    }));
+
+    function promiseFn(arg) {
+      assert.strictEqual(arg, value);
+      return Promise.resolve(arg);
+    }
+
+    const cbPromiseFn = callbackify(promiseFn);
+
+    cbPromiseFn(value, common.mustCall((err, ret) => {
+      assert.ifError(err);
+      assert.strictEqual(ret, value);
+    }));
+  }
+}
+
+{
+  // Test that `this` binding is the same for callbackified and original
+  for (const value of values) {
+    const iAmThis = {
+      fn(arg) {
+        assert.strictEqual(this, iAmThis);
+        return Promise.resolve(arg);
+      },
+    };
+    iAmThis.cbFn = callbackify(iAmThis.fn);
+
+    iAmThis.cbFn(value, common.mustCall(function(err, ret) {
+      assert.ifError(err);
+      assert.strictEqual(ret, value);
+      assert.strictEqual(this, iAmThis);
+    }));
+
+    const iAmThat = {
+      async fn(arg) {
+        assert.strictEqual(this, iAmThat);
+        return await Promise.resolve(arg);
+      },
+    };
+    iAmThat.cbFn = callbackify(iAmThat.fn);
+
+    iAmThat.cbFn(value, common.mustCall(function(err, ret) {
+      assert.ifError(err);
+      assert.strictEqual(ret, value);
+      assert.strictEqual(this, iAmThat);
+    }));
+  }
+}
+
+{
+  // Test that callback that throws emits an `uncaughtException` event
+  const fixture = join(fixtureDir, 'callbackify1.js');
+  execFile(
+    process.argv[0],
```

Comment:
Nit: Maybe `process.execPath` would be clearer than `process.argv[0]`, but it doesn't matter much either way.